Wrappers/Python/cil/optimisation/algorithms/SPDHG.py

```diff
@@ -124,100 +314,91 @@ def set_up(self, f, g, operator, tau=None, sigma=None, \
             List of Step size parameters for Dual problem
         initial : DataContainer, optional, default=None
             Initial point for the SPDHG algorithm
-        prob : list of floats, optional, default=None
-            List of probabilities. If None each subset will have probability = 1/number of subsets
         gamma : float
             parameter controlling the trade-off between the primal and dual step sizes
-
-        **kwargs:
-        norms : list of floats
-            precalculated list of norms of the operators
+        sampler: an instance of a `cil.optimisation.utilities.Sampler` class
+             Method of selecting the next index for the SPDHG update. If None, random sampling and each index will have probability = 1/number of subsets
         '''
         logging.info("{} setting up".format(self.__class__.__name__, ))
-                    
+
         # algorithmic parameters
         self.f = f
         self.g = g
         self.operator = operator
-        self.tau = tau
-        self.sigma = sigma
-        self.prob = prob
-        self.ndual_subsets = len(self.operator)
-        self.gamma = gamma
-        self.rho = .99
-        
-        if self.prob is None:
-            self.prob = [1/self.ndual_subsets] * self.ndual_subsets
-
-        
-        if self.sigma is None:
-            if norms is None:
-                # Compute norm of each sub-operator       
-                norms = [operator.get_item(i,0).norm() for i in range(self.ndual_subsets)]
-            self.norms = norms
-            self.sigma = [self.gamma * self.rho / ni for ni in norms] 
-        if self.tau is None:
-            self.tau = min( [ pi / ( si * ni**2 ) for pi, ni, si in zip(self.prob, norms, self.sigma)] ) 
-            self.tau *= (self.rho / self.gamma)
-
-        # initialize primal variable 
+        self.ndual_subsets = self.operator.shape[0]
+        self.sampler = sampler
+        self._deprecated_kwargs(deprecated_kwargs)
```

Comment:
I think this is fine. self._deprecated_kwargs uses the sampler passed to the setup (which could be None). If deprecated `prob`` is passed than self._deprecated_kwargs will set up a sampler and if not then line below will set up a sampler. It has been designed so that the function and the call can be deleted once we deprecate the arguments. 
```python
self._sampler = sampler
self._deprecated_kwargs(deprecated_kwargs)
self._sampler is None:
    self._sampler = Sampler.random_with_replacement(len(operator))
```